concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/BlockSummary.java

```diff
@@ -12,10 +21,33 @@
 @Getter
 @ToString
 public final class BlockSummary {
-    private final List<Object> transactionSummaries;
-    private final List<SpecialEvent> specialEvents;
+    /**
+     * The protocol version of the chain at the block.
+     */
+    private final int protocolVersion;
+    /**
+     * The list of transaction summaries for the block.
+     */
+    private final List<TransactionSummary> transactionSummaries;
```

Comment:
What do you mean? Like a `Map Hash TransactionSummary` etc?

Reply:
The List interface guarantees that you have access by index, but it might be that this access is linear in the index.

It would be better to guarantee that it is constant-time if possible. Don't know if there is such an interface. If not, why not just use the concrete ArrayList type?

Reply:
Correct. The `jackson` json library defaults to `ArrayList` when deserializing lists. which as you already pointed out is optimal here. 

I agree that in order to get an unpleasant surprise at some point in the future then they should probably be `ArrayList`s both of them explicitly.